common/user.js

```diff
@@ -617,6 +617,10 @@ class User {
 
     req.body.roles ??= [];
 
+    if (!User.hasValidRoles(req.body.roles)) {
```

Comment:
Should we error out like this or should we just silently remove the "bad" roles and still update/create the user?

Reply:
I like the explicit approach of erroring, since it would allow the requester to realize their mistake instead of it going unnoticed. And I imagine this isn't a problem someone would run into on the UI (since we use RoleDropdown that only give valid roles as options), so it shouldn't hurt usability 🙂

Reply:
agree